model_utils/models.py

```diff
@@ -32,10 +32,8 @@ def save(self, *args, **kwargs):
         a parameter to the update field argument.
         """
         update_fields = kwargs.get('update_fields', None)
-        if update_fields is not None:
-            update_fields = set(update_fields)
-            if update_fields:
-                kwargs['update_fields'] = update_fields.union({'modified'})
+        if update_fields and 'modified' not in update_fields:
+            kwargs['update_fields'] = list(kwargs['update_fields']) + ['modified']
```

Comment:
I think the previous approach was fine.

```suggestion
        if update_fields:
            kwargs['update_fields'] = set(update_fields).union({'modified'})
```

Reply:
Thanks @charettes  I changed it to set approach